### ISSUE TITLE
[4.3] Scene Dock does not open Orchestration Script

### ISSUE THREAD

**Issue by Naros** [Vahera/godot-orchestrator#77]
### Describe the bug

When using Godot 4.3.dev2, the attached orchestration script is not opened in the Orchestrator tab.  

### Expected behavior

Click the Scrpt icon opens the Orchestration.

### Actual behavior

No action taken.

### How to Reproduce?

1. Download Godot 4.3.dev2
2. Attach an Orchestration to a scene node.
3. Save all files and close the Orchestration.
4. Click the script icon on the node, nothing happens.

### Godot full version

4.3.dev2

### Orchestrator version

2.0.dev1

### Additional information

Attaching a `GDScript` seems to work.

**Comment by Naros** [comment]
This specific issue is a direct change in behavior on Godot 4.3, specifically because:
```cpp
void SceneTreeDock::_script_open_request(const Ref<Script> &p_script) {
	EditorNode::get_singleton()->edit_resource(p_script);
}
```
behavior was changed to:
```cpp
void SceneTreeDock::_script_open_request(const Ref<Script> &p_script) {
	if (ScriptEditor::get_singleton()->edit(p_script, true)) {
		EditorNode::get_singleton()->editor_select(EditorNode::EDITOR_SCRIPT);
	}
}
```
This change explicitly mandates that any script plug-in is forced to implement the `ScriptEditorBase` contracts, which we do not implement for various reasons.  We will need to reach out to the editor team and understand whether this will be mandatory moving forward or whether or not we can revert this change in a future 4.3 build.  If this is mandatory, we'll need to rewrite a vast majority of the integration with the editor :(

**Comment by Naros** [comment]
Opened https://github.com/godotengine/godot/pull/87722 in upstream Godot.

**Comment by Naros** [comment]
This was fixed in Godot 4.3.dev3.
